### PR TITLE
Bundle AlignedMemory inputs with MemoryBundle

### PR DESCRIPTION
**Pull request by qianqianzhu**
This PR addresses https://github.com/browsermt/bergamot-translator/issues/133 (2) and (3) which aims to bundle all byte-array members together.

### CODE REVIEW

**Review by XapaJIaMnu — DISMISSED** [review]

**Comment by qianqianzhu** [comment]
@abhi-agg Service API is changed here thus WASM builds failed. Shall I leave the fix for you?

**Comment by XapaJIaMnu** [comment]
Fix before merge

File: src/translator/definitions.h [hunk 1]
```diff
@@ -15,6 +15,13 @@ typedef std::vector<Segment> Segments;
 /// Shortcut to AlignedVector<char> for byte arrays
 typedef AlignedVector<char> AlignedMemory;
 
+struct MemoryBundle {
+  AlignedMemory model;
+  AlignedMemory shortlist;
+  std::vector<std::shared_ptr<AlignedMemory>> vocabs;
+  AlignedMemory ssplitPrefixFile;
+};
```
Comment (jerinphilip) [inline src/translator/definitions.h:23]: Move documentation from Service class here. All the aligned to X, what the vocab represents bits.

File: src/translator/service.h [hunk 2]
```diff
@@ -94,14 +78,8 @@ class Service {
   /// @param [in] vocabMemories vector of vocabulary memories (aligned to 64). Optional.
```
Comment (jerinphilip) [inline src/translator/service.h:78]: Update docs.

**Review by jerinphilip — CHANGES_REQUESTED** [review]
@qianqianzhu Do magic here to get green. PR shouldn't go in without green.

https://github.com/browsermt/bergamot-translator/blob/d7cb859ab758dd0b04f1dc8ddeda03216a319c2b/wasm/bindings/TranslationModelBindings.cpp#L51-L58

File: src/translator/service.h [hunk 3]
```diff
@@ -67,23 +67,7 @@ class Service {
...
-  ///
-  /// This is equivalent to a call to:
-  /// ```cpp
-  ///    Service(options, AlignedMemory(), AlignedMemory(), {})
-  /// ```
-  /// wherein empty memory is passed and internal flow defaults to file-based
-  /// model, shortlist loading. AlignedMemory() corresponds to empty memory
-  explicit Service(Ptr<Options> options)
-      : Service(options, AlignedMemory(), AlignedMemory(), {}) {}
+  explicit Service(Ptr<Options> options, MemoryBundle memoryBundle={});
```
Comment (abhi-agg) [inline src/translator/service.h:70]: Could you please update the documentation of all the 3 constructors of this class?

Comment (qianqianzhu) [reply src/translator/service.h:70]: Yes, I will do (trying to fix the WASM build first)

Comment (abhi-agg) [reply src/translator/service.h:70]: @qianqianzhu Please let me know if you need help there 👍

**Review by jerinphilip — APPROVED** [review]

**Review by XapaJIaMnu — APPROVED** [review]

**Pull request merged**